### DISCUSSION TITLE
Trying to fix a java.net.SocketTimeoutException

### DISCUSSION THREAD

**Discussion by markstubbs** [keycloak/keycloak#13309] [Q&A] [answered]
Hi - we're using our WildFly-based Keycloak (v.16.1.1) to talk upstream to a 3rd-party IdentityProvider. It mostly works but we're seeing a number of "Failed to make identity provider oauth callback: java.net.SocketTimeoutException: Read timed out" errors in our logs. In order to remediate this we've tried to change the timeout value by setting this in our standalone.xml file:

`            <spi name="connectionsHttpClient">
                <provider name="default" enabled="true">
                        <properties>
                                <property name="socket-timeout-millis" value="15000" />
                        </properties>
                </provider>
            </spi>
`
But despite this we're still seeing timeouts. I'm not a WildFly expert so my question is, is this the correct place to configure this value? If so, is there some command I can use to check that WildFly/Keycloak is actually using it?  Thanks.

**Comment by sandeepmachiraju** [comment]
Found any solution for this?

**Comment by markstubbs** [comment, marked as answer]
In the end we added this to our `standalone.xml` file:

```
 <subsystem xmlns="urn:jboss:domain:keycloak-server:1.1">
            ...
            <spi name="connectionsHttpClient">
                <provider name="default" enabled="true">
                        <properties>
                                <property name="socket-timeout-millis" value="30000" />
                        </properties>
                </provider>
            </spi>
             ...
</subsystem>
```
which seemed to fix it.

**Reply by sandeepmachiraju** [reply]
I added it, but still, I am seeing the problem.

**Reply by sandeepmachiraju** [reply]
It's working now. My sed command is busted while replacing the content mentioned above in Dockerfile. For anyone who is trying to use dockerfile (for keycloak older versions), please refer to this link: https://stackoverflow.com/a/73546541/6553708

**Comment by kherock** [comment]
I had the same question for Quarkus distributions and found my answer using help from the answer here and from this documentation:
https://www.keycloak.org/server/configuration-provider

Using the CLI, you can pass `--spi-connections-http-client-default-socket-timeout-millis=30000`. 

I haven't tested how this can be implemented by `keycloak.conf`, but it looks like the following should work based on [these docs](https://github.com/keycloak/keycloak-community/blob/main/design/keycloak.x/configuration.md#spis-and-provider-configuration).

```
# keycloak.conf

spi.connections-http-client.default.socket-timeout-millis=30000
```

For anyone interested in what the default value is, It appears that it's a 5 second timeout set here:
https://github.com/keycloak/keycloak/blob/45b203d74c267de60706d8dbfee0ef06a6e37336/services/src/main/java/org/keycloak/connections/httpclient/DefaultHttpClientFactory.java#L146
